wpilibj/src/main/java/edu/wpi/first/wpilibj/LEDPattern.java

```diff
@@ -145,7 +145,29 @@ default <T extends LEDReader & LEDWriter> void applyTo(T readWriter) {
   default LEDPattern reversed() {
     return (reader, writer) -> {
       int bufLen = reader.getLength();
-      applyTo(reader, (i, r, g, b) -> writer.setRGB((bufLen - 1) - i, r, g, b));
+      applyTo(
+          new LEDReader() {
```

Comment:
A more generic LEDReader implementation that accepts an arbitrary remapping function may be better than a single-purpose class for offsets and this anonymous class

Reply:
Going a step further, would it be possible to have an overload that accepts a remapping function that is applied to both the reader and writer?

Reply:
Made a PR that adds a `mapIndex()` decorator that applies the mapping function to both the reader and writer: https://github.com/SamCarlberg/allwpilib/pull/2